nbgrader/apps/extensionapp.py

```diff
@@ -1,299 +1,35 @@
-        extra_args = self.extra_args[:]
-
-        # install the create_assignment extension
-        if len(extra_args) == 0 or "create_assignment" in extra_args:
-            self.log.info("Installing create_assignment extension")
-            self.extra_args = [os.path.join(nbextensions_dir, 'static', 'create_assignment')]
-            self.install_extensions()
-
-        # install the assignment_list extension
-        if sys.platform != 'win32' and (len(extra_args) == 0 or "assignment_list" in extra_args):
-            self.log.info("Installing assignment_list extension")
-            self.extra_args = [os.path.join(nbextensions_dir, 'static', 'assignment_list')]
-            self.install_extensions()
-
-
-class ExtensionUninstallApp(UninstallNBExtensionApp, NbGrader):
-
-    name = u'nbgrader-extension-uninstall'
-    description = u'Uninstall the nbgrader extensions'
-
-    examples = """
-
-    To uninstall all the nbgrader extensions that are installed systemwide,
-    run:
-
-        nbgrader extension uninstall
-
-    If you want to uninstall the extensions installed in your user
-    environment, use:
-
-        nbgrader extension uninstall --user
-
-    To uninstall only a specific extension, you can pass the name of the
-    extension you want to uninstall as an argument, e.g.:
-
-        nbgrader extension uninstall create_assignment
-        nbgrader extension uninstall assignment_list
-
-    """
-
-    destination = Unicode('')
-
-    def _classes_default(self):
-        return [ExtensionUninstallApp, UninstallNBExtensionApp]
-
-    def excepthook(self, etype, evalue, tb):
-        format_excepthook(etype, evalue, tb)
-
-    def start(self):
-        nbextensions_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'nbextensions'))
-        extra_args = self.extra_args[:]
-
-        # install the create_assignment extension
-        if len(extra_args) == 0 or "create_assignment" in extra_args:
-            self.log.info("Uninstalling create_assignment extension")
-            self.extra_args = ["create_assignment"]
-            self.uninstall_extensions()
-
-        # install the assignment_list extension
-        if sys.platform != 'win32' and (len(extra_args) == 0 or "assignment_list" in extra_args):
-            self.log.info("Uninstalling assignment_list extension")
-            self.extra_args = ["assignment_list"]
-            self.uninstall_extensions()
-
-
-class ExtensionActivateApp(EnableNBExtensionApp, NbGrader):
+_compat_message = """
+The installation of the nbgrader extensions are now managed through the
+`jupyter nbextension` and `jupyter serverextension` commands.
 
-    name = u'nbgrader-extension-activate'
-    description = u'Activate the nbgrader extension'
+To install and enable the nbextensions (assignment_list and create_assignment) run:
 
-    flags = {}
-    aliases = {}
+    $ jupyter nbextension install --sys-prefix --py nbgrader
```

Comment:
The direction we are moving is to keep nbextension/jupyterlab extensions well separated from the server extensions. A couple of reasons for this:

* A single server could be serving notebooks to multiple frontends (notebook, nteract, jupyterlab)
* We are in the process of separating the notebook server into its own project and repo that won't share any code with the frontends
* We are also moving to a different approach for installing all of these things that will make it all easier for users. While we do this, we want to avoid making things more complicated with our existing stuff.

Obviously, in nbgrader, we could wrap all that and expose an endpoint that does it all automatically. For that I think the best approach is to create a conda-forge feedstock with post install scripts that do everything. This is what we are doing with the vega package and the install process is then trivial (no post install commands needed). Are you ok with that approach? Here is the vega-feedstock that has the needed scripts:

https://github.com/conda-forge/vega-feedstock